weblab_cg/_chaste_printer.py

```diff
@@ -0,0 +1,91 @@
+from ._printer import WebLabPrinter
+import sympy
+
+from sympy.printing.precedence import precedence
+
+
+class ChastePrinter(WebLabPrinter):
+    """
+    Converts Sympy expressions to strings for use in Chaste code generation, based on the WeblabPrinter.
+
+    To use, create a :class:`ChastePrinter` instance, and call its method
+    :meth:`doprint()` with a Sympy expression argument.
+
+    Arguments:
+
+    ``symbol_function``
+        A function that converts symbols to strings (variable names).
+    ``derivative_function``
+        A function that converts derivatives to strings.
+
+    """
+
+    def __init__(self, symbol_function=None, derivative_function=None):
+        super().__init__(symbol_function, derivative_function)
+
+        # Do not use the math. prefix for functions, as chaste c++ files include an include for this
+        self._prefix = ''
+
+        # Make sure we can output a call to GetIntracellularAreaStimulus
+        WebLabPrinter._math_functions.add('GetIntracellularAreaStimulus')
```

Comment:
Bit confused about this. `_math_functions` is a list of functions that exist in both sympy and python's `math` module, and so can be passed through directly. (1) these names are not necessarily the same in C++ and (2) there's certainly no GetIntracellularAreaStimulus in sympy

Reply:
Indeed that's why I added them! Maybe math_functions is the wrong name and we should change it to just functions. If I don't add these in as recognized functions the printer fails when I try to output them (telling me i'm writing out an unknown function).

Reply:
Hmmmm, but there's no sympy function that maps to GetIntracellularAreaStimulus, is there?

Reply:
As far as I understand you're just defining functions that are legal in the output. They don't have to by in sympy

Reply:
It's used exclusively by `_print_Function(self, expr):`, which is called when a sympy function needs to be printed

Reply:
Fine we'll call it _custom_functions. It's used in `_print_Function` to be able to print function calls